include/power/power_state.h

```diff
@@ -103,6 +103,14 @@ enum pm_state {
 	PM_STATE_SOFT_OFF
 };
 
+/**
+ * Information about a power management state
+ */
+struct pm_state_info {
+	enum pm_state state;
+	int32_t min_residency_us;
```

Comment:
After some thought, I think we should add all values, when you actually do the profiling and get the data for min-residency you probably want to add the other values as well, basically entry, exit, wakeup etc., the same way Linux does it.

In current implementation we only check for min-residency, but if someone writes their own policy, they could use other data depending on the use-case. 

Do not see any overhead in adding that and supporting it right now.

Reply:
Yep, I agree that we will end up having them. But currently I don't even have reference values for these things ... I am changing the current code with this and thought add unused  information now would just bring noise to that.

Can we keep it for now, and add these properties later ?

Reply:
not sure why you need reference values, this should be in the bindings and for now, if they are not filled you do not need to do anything, it is not like we have reference for min-residency-us either, this will need to be done based on someone actually knowing their hardware and profiling mostly. For testing we can just make up those values.

Reply:
I mean, I have residency time for a target `mec15xxevb_assy6853` that I am feeding in this board dts in my changes, but these other properties have no current use, we currently don't even have a case where I'll use it, so what I'm proposing is adding these properties after the code change.

Just add these properties in the binding is no problem, but then I need to add these properties in the pm_state_info and a lot of unused fields may be confuse during these transition.

Of course, if that is a blocker I can just add it now, it is very easy to do it.

Reply:
I just don't particularly like the idea of having data the usage is not 100% clear yet. It is quite easy to forget removing them later if they are not needed ...